sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/BufferStagingArea.java

```diff
@@ -79,25 +79,50 @@ public Flux<BufferAggregator> write(ByteBuffer buf) {
                 result = Flux.empty();
             }
         } else {
-            // We will overflow the current buffer and require another one.
-            // Duplicate and adjust the window of buf so that we fill up currentBuf without going out of bounds.
+            // When the incoming buffer size exceeds the remaining capacity of the current buffer,
+            // we need to handle overflow by creating new BufferAggregators.
+
+            // Duplicate the buffer to manipulate its position and limit without affecting the original buffer.
             ByteBuffer duplicate = buf.duplicate();
+            // Calculate the new limit for the duplicate buffer to exactly fill up the current buffer.
             int newLimit = buf.position() + (int) this.currentBuf.remainingCapacity();
             duplicate.limit(newLimit);
             this.currentBuf.append(duplicate);
             // Adjust the window of original buffer to represent remaining part.
             buf.position(newLimit);
-
+            // Add the current buffer to the result as it is now full.
             result = Flux.just(this.currentBuf);
 
-            /*
-            Get a new buffer and fill it with whatever is left from buf. Note that this relies on the assumption that
-            the source Flux has been split up into buffers that are no bigger than chunk size. This assumption
-            means we'll only have to over flow once, and the buffer we overflow into will not be filled. This is the
-            buffer we will write to on the next call to write().
-             */
-            this.currentBuf = new BufferAggregator(this.buffSize);
-            this.currentBuf.append(buf);
+            // Calculate the number of complete chunks that can be formed from the remaining data.
+            int remainingChunks = buf.remaining() / (int) this.buffSize;
+            if (remainingChunks >= 1) {
+                // If there are complete chunks to be processed, create new BufferAggregators for each.
+                BufferAggregator[] aggregators = new BufferAggregator[remainingChunks];
+                for (int i = 0; i < remainingChunks; i++) {
+                    BufferAggregator overflowBuffer = new BufferAggregator(this.buffSize);
```

Comment:
To help readability here, could we call this `aggregator`. Trying to read this with both `overflowBuffer` and `overflowDup` is a bit hard.